Cauli/CauliURLProtocol/InterceptingInputStream.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  InterceptingInputStream.swift
+//  Cauliframework
+//
+//  Created by Cornelius Horstmann on 19.02.19.
+//
+
+import Foundation
+
+internal protocol InterceptingInputStreamDelegate {
+    func didRead(_ buffer: UnsafeMutablePointer<UInt8>, maxLength len: Int)
+}
+
+internal class InterceptingInputStream: InputStream {
```

Comment:
Super minor, but I think that class should be final, what do you think?

Reply:
Yes, you are correct. Will change this.